packages/web/src/pages/sign-up-page/pages/CreateEmailPage.tsx

```diff
@@ -0,0 +1,223 @@
+          }
+        } catch (e) {
+          // Unknown error state ¯\_(ツ)_/¯
+          setErrors({ email: messages.unknownError })
+        }
+      }
+    },
+    [dispatch, navigate, queryContext]
+  )
+
+  return (
+    <Formik
+      validationSchema={FormSchema}
+      initialValues={initialValues}
+      onSubmit={onSubmit}
+      validateOnBlur
+      validateOnChange={false}
+    >
+      {({ isSubmitting }) => (
+        <Flex
+          as={Form}
+          direction='column'
+          h='100%'
+          w='100%'
+          ph={isMobile ? 'l' : '2xl'}
+          pv='2xl'
+          gap='2xl'
+        >
+          <Box alignSelf='center'>
+            {isMobile ? (
+              <IconAudiusLogoHorizontalColor />
+            ) : (
+              <PreloadImage
+                src={audiusLogoColored}
+                alt='Audius Colored Logo'
+                css={{
+                  height: 160,
+                  width: 160,
+                  objectFit: 'contain'
+                }}
+              />
+            )}
+          </Box>
+          <Flex direction='column' gap={isMobile ? 's' : 'l'}>
+            <Text
+              variant='heading'
+              size={isMobile ? 'm' : 'l'}
+              color='accent'
+              tag='h1'
+              css={{ textAlign: isMobile ? 'center' : undefined }}
+            >
+              {messages.title}
+            </Text>
+            <Text
+              color='default'
+              size={isMobile ? 'm' : 'l'}
+              variant='body'
+              tag='h2'
+              css={{ textAlign: isMobile ? 'center' : undefined }}
+            >
+              {messages.subHeader}
+            </Text>
+          </Flex>
+          <Flex direction='column' gap='l'>
+            <HarmonyTextField
+              name='email'
+              autoComplete='email'
+              label={messages.emailLabel}
+            />
+            <Divider css={{ width: '100%' }}>
+              <Text variant='body' size={isMobile ? 's' : 'm'} color='subdued'>
+                {messages.socialsDividerText}
+              </Text>
+            </Divider>
+            <SocialMediaLoginOptions
+              onCompleteSocialMediaLogin={handleLinkedSocialMedia}
+            />
+          </Flex>
+          <Flex direction='column' gap='l'>
+            <Button
+              variant={ButtonType.PRIMARY}
+              type='submit'
+              fullWidth
+              iconRight={IconArrowRight}
+              isLoading={isSubmitting}
+            >
+              {messages.signUp}
+            </Button>
+
+            <Text
+              variant='body'
+              size={isMobile ? 'm' : 'l'}
+              css={{ textAlign: isMobile ? 'center' : undefined }}
+            >
+              {messages.haveAccount}{' '}
+              <TextLink variant='visible' asChild>
+                <Link to={SIGN_IN_PAGE}>{messages.signIn}</Link>
+              </TextLink>
```

Comment:
It feels like this should be possible in a single component. Maybe we want a common set of Harmony wrappers for web?

`import TextLink from 'src/harmony-wrapped/link` or something which uses react-router link internally?

Reply:
yeah would def welcome this, cause it'll be a really common use-case

Reply:
id be open to us creating a few helper components, both for entire app, and for sign-up, since there are lot of repeated styles on each page.